From c15c3efd31be1bca130e3ec95e2932f8dd8ed1b1 Mon Sep 17 00:00:00 2001
From: Michael Adam <obnox@samba.org>
Date: Mon, 16 May 2016 01:39:09 +0200
Subject: [PATCH 1080/1096] s3:vfs: add 'kernel_share_modes_taken' to
 files_struct

This will allow to track whether kernel share modes have been
taken at open and correclty remove them again on close.

BUG: https://bugzilla.samba.org/show_bug.cgi?id=11919

Signed-off-by: Michael Adam <obnox@samba.org>
Reviewed-by: Christian Ambach <ambi@samba.org>
(cherry picked from commit 356487345724ce5dffdddf9c60735b2c965f30bc)
(cherry picked from commit 68249a719ae99e3b23d098f7494a1c1f23225659)
---
 source3/include/vfs.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/source3/include/vfs.h b/source3/include/vfs.h
index c18ea59..fb3b8aa 100644
--- a/source3/include/vfs.h
+++ b/source3/include/vfs.h
@@ -237,6 +237,8 @@ typedef struct files_struct {
 	uint32_t access_mask;		/* NTCreateX access bits (FILE_READ_DATA etc.) */
 	uint32_t share_access;		/* NTCreateX share constants (FILE_SHARE_READ|FILE_SHARE_WRITE|FILE_SHARE_DELETE). */
 
+	bool kernel_share_modes_taken;
+
 	bool update_write_time_triggered;
 	struct tevent_timer *update_write_time_event;
 	bool update_write_time_on_close;
-- 
2.9.0

